s_algolia: fix facet settings export

simahawk opened the pull request.

simahawk committed:
s_algolia: fix facet settings export

Settings must be exported by lang

guewen commented on shopinvader_algolia/models/shopinvader_product.py:
```diff
@@ -44,9 +46,22 @@ def _get_facetting_values(self, se_bakend):
+        # to unify them in indexes and Locomotive settings.
+        filters = invader_backend.with_context(lang=lang.code).filter_ids
+        # Avoid value cached by previous call w/ another lang
+        filters.invalidate_cache(["display_name"])
```
I don't know if it's relevant here, but maybe a `depends_context("lang")` could be added on the display_name method to fix this globally.

simahawk replied:
you are right, I forgot we had this gem in 13.0 :sweat_smile:  Thank you Mister Memory! :smile: 
Refactored w/ it, works like a charm :smirk:

simahawk committed:
product.filter: fix _compute_display_name by lang

Filter values depend on language (prod attr name, field name, etc).
Before this change you could get wrong values when computing search engine indexes
which are linked to specific languages.

simahawk commented:
/ocabot merge patch
